src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1782,6 +1782,17 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
                 return;
             }
 
+            // Note: reject requests whose client is not a known audience.
+            if (notification.Audiences.Contains(clientNotification.ClientId)) {
```

Comment:
Oops, yeah, forgot to negate it. Other than that I wasn't sure how else to determine the audience.

-----Original Message-----
From: "Kévin Chalet" <notifications@github.com>
Sent: ‎10/‎27/‎2015 6:10 PM
To: "aspnet-contrib/AspNet.Security.OpenIdConnect.Server" <AspNet.Security.OpenIdConnect.Server@noreply.github.com>
Cc: "Michael Ciarlillo" <MonkeyJamboree@gmail.com>
Subject: Re: [AspNet.Security.OpenIdConnect.Server] Reject introspectionrequests whose client is not a known audience #156 (#159)

In src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs:
> @@ -1782,6 +1782,17 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
>                  return;
>              }
>  
> +            // Note: reject requests whose client is not a known audience.
> +            if (notification.Audiences.Contains(clientNotification.ClientId)) {
Missing !?  
Note that this check won't work if the client was not authenticated.
—
Reply to this email directly or view it on GitHub.